contracts/test/walletAction-test.ts

```diff
@@ -288,6 +288,123 @@ describe("WalletActions", async function () {
     await expect(fx.verificationGateway.callStatic.verify(tx)).to.be.rejected;
   });
 
+  it("(ERC-1271) should verify message", async function () {
+    const wallet = await fx.createBLSWallet();
+    const blsWallet = await ethers.getContractAt("BLSWallet", wallet.address);
+
+    const hashedMessage = ethers.utils.keccak256(
+      ethers.utils.toUtf8Bytes("I'm a message"),
+    );
+
+    const signature = wallet.signMessage(hashedMessage);
+    const concatenatedSignature =
+      ethers.utils.hexlify(signature[0]) +
+      ethers.utils.hexlify(signature[1]).substring(2);
+    const signatureBytes = ethers.utils.arrayify(concatenatedSignature);
+
+    const expectedBytes4MagicValue = ethers.utils
+      .solidityKeccak256(["string"], ["isValidSignature(bytes32,bytes)"])
+      .substring(0, 10);
+
+    // initial call needed to ensure isValidSignature is successful
+    await fx.call(wallet, blsWallet, "nonce", [], 0, 30_000_000);
```

Comment:
Had to run this call in order to get the tests working, otherwise an exception was thrown on `blsWallet.isValidSignature`. Wasn't entirely sure on the underlying reason it works. Wallet initialisation?